### PR TITLE
Can create subgroups containing users without usernames

### LINKED ISSUE #38: Handle creation of subgroups when a user does not have a username
We can `text_markdown_v2_urled` property to extract the `telegram_user_id`

```
'/group test [John](tg://user?id=12345678)'
```

### PR DESCRIPTION
**Pull request by AshciR**
fixes #38

### CODE REVIEW

File: src/shout_subgroup/models.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ class UserModel(Base):
     __tablename__ = 'users'
     user_id = Column(String, primary_key=True, default=lambda: str(uuid4()))
     telegram_user_id = Column(Integer, nullable=False, unique=True)
-    username = Column(String, nullable=False)
+    username = Column(String, nullable=True)
```
Comment (AshciR) [inline src/shout_subgroup/models.py:30]: Telegram usernames can be empty

File: src/shout_subgroup/utils.py [hunk 2]
```diff
@@ -52,3 +58,78 @@ async def format_telegram_usernames(usernames: set[str], telegram_user: User) ->
 def _replace_me_mention_with_username(username: str, telegram_user: User) -> str:
     # TODO: Update logic when we decide how to handle Users without usernames
     return telegram_user.username.lower() if username.lower() == "me" else username
+
+
+@dataclass(frozen=True, eq=True)
+class UserIdMentionMapping:
```
Comment (AshciR) [inline src/shout_subgroup/utils.py:64]: @rick1e Ended up doing this. We need a way to map the mention to the user id

File: src/shout_subgroup/utils.py [hunk 3]
```diff
@@ -52,3 +58,78 @@ async def format_telegram_usernames(usernames: set[str], telegram_user: User) ->
...
+    """
+    Convenient class for tying user ids to the mentions
+    """
+    mention: str
+    user_id: str | None
+
+
+async def get_user_id_from_mention(db: Session, username_or_markdown: str) -> UserIdMentionMapping:
```
Comment (AshciR) [inline src/shout_subgroup/utils.py:72]: Focus

File: src/shout_subgroup/utils.py [hunk 4]
```diff
@@ -52,3 +58,78 @@ async def format_telegram_usernames(usernames: set[str], telegram_user: User) ->
...
+    :return: the user id used within our system. None if user doesn't exist
+    """
+    user = await find_user_by_username(db, telegram_username)
+    if not user:
+        return None
+
+    return user.user_id
+
+
+async def _convert_markdown_to_user_id(db: Session, telegram_markdown_v2: str) -> int | None:
```
Comment (AshciR) [inline src/shout_subgroup/utils.py:96]: Focus

**Review by rick1e — APPROVED** [review]

**Pull request merged**